compiler/hash-typecheck/src/new/passes/state.rs

```diff
@@ -0,0 +1,102 @@
+//! Helper structs for keeping track of state during AST traversal.
```

Comment:
You could move this into `utils`, I am sure there are other places that benefit this.